Userland/Libraries/LibCore/EventLoop.cpp

```diff
@@ -74,7 +74,7 @@ thread_local bool EventLoop::s_wake_pipe_initialized { false };
 void EventLoop::initialize_wake_pipes()
 {
     if (!s_wake_pipe_initialized) {
-#if defined(SOCK_NONBLOCK)
+#if defined(SOCK_NONBLOCK) && !defined(AK_OS_WIN32)
```

Comment:
er, what's going on here? the first check should be for O_CLOEXEC. then you can remove the WIN32 check

Reply:
This should be filed as a separate issue. Also, the cygwin check would still be required since `O_CLOEXEC` exists under cygwin but not `pipe2`.